plenum/server/consensus/ordering_service.py

```diff
@@ -2310,6 +2310,26 @@ def process_view_change_started(self, msg: ViewChangeStarted):
 
     def process_new_view_accepted(self, msg: NewViewAccepted):
         self._write_manager.future_primary_handler.set_primaries(msg.view_no, self._data.primaries)
+        self._setup_for_non_master_after_view_change(msg.view_no)
+
+    def _setup_for_non_master_after_view_change(self, current_view):
+        if not self.is_master:
+            for v in list(self.stashed_out_of_order_commits.keys()):
+                if v < current_view:
+                    self.stashed_out_of_order_commits.pop(v)
+
+    def process_catchup_finished(self, msg: CatchupFinished):
+        if compare_3PC_keys(msg.master_last_ordered,
+                            msg.last_caught_up_3PC) > 0:
+            if self.is_master:
+                self._caught_up_till_3pc(msg.last_caught_up_3PC)
+            else:
+                self.first_batch_after_catchup = True
+                self.catchup_clear_for_backup()
+        # This method was in primaryName setter
+        self.gc(msg.last_caught_up_3PC)
```

Comment:
Should we call it for `last_caught_up_3PC`, or for the checkpoint closest to the `last_caught_up_3PC`?

Reply:
It is not needed here at all, because CheckpointService sends CheckpointStabilized message, which is handled by OrderingService by calling `self.gc(stable_checkpoint)` (not exactly, but meaning is the same).